### ISSUE TITLE
TypeError when using ClassAssistedBuilder with @inject

### ISSUE THREAD

**Issue by davidparsson** [alecthomas/injector#63]
I've been trying to use the `ClassAssistedBuilder` and the `AssistedBuilder` as described in [the docs](http://injector.readthedocs.io/en/latest/terminology.html#assisted-injection), but I'm getting the following exception:
```
Traceback (most recent call last):
  File "reproduce.py", line 9, in <module>
    class NeedsUserUpdater(object):
  File "reproduce.py", line 11, in NeedsUserUpdater
    def __init__(self, builder: ClassAssistedBuilder[UserUpdater]):
  File "/Users/me/injector-reproduce/.venvs/3.5/lib/python3.5/site-packages/injector.py", line 1093, in inject
    bindings = _infer_injected_bindings(function)
  File "/Users/me/injector-reproduce/.venvs/3.5/lib/python3.5/site-packages/injector.py", line 967, in _infer_injected_bindings
    if _is_specialization(v, Union):
  File "/Users/me/injector-reproduce/.venvs/3.5/lib/python3.5/site-packages/injector.py", line 490, in _is_specialization
    (cls.__origin__ is generic_class or issubclass(cls.__origin__, generic_class))
  File "/usr/local/opt/pyenv/versions/3.5.3/lib/python3.5/typing.py", line 770, in __subclasscheck__
    raise TypeError("Unions cannot be used with issubclass().")
TypeError: Unions cannot be used with issubclass().
```

This is the code I'm running:

```
#!/usr/bin/env python3
from injector import ClassAssistedBuilder, Injector, inject


class UserUpdater:
    def __init__(self, user):
        self._user = user

class NeedsUserUpdater(object):
    @inject
    def __init__(self, builder: ClassAssistedBuilder[UserUpdater]):
        self.updater_builder = builder

    def method(self):
        updater = self.updater_builder.build(user=None)


injector = Injector()
instance = injector.get(NeedsUserUpdater)
instance.method()
```

Am I doing something wrong, or is it not working as intended?

**Comment by davidparsson** [comment]
It however works when the `@inject` decorator is not used:
```
injector = Injector()
builder = injector.get(ClassAssistedBuilder[UserUpdater])
instance = NeedsUserUpdater(builder)
instance.method()
```

**Comment by m3drano** [comment]
I believe that PR #61 fixes this, but it has not yet been released.

**Comment by davidparsson** [comment]
That looks probable, yes. Thanks!

**Comment by alecthomas** [comment]
Just released 0.12.1 which includes this fix.
